var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -369,7 +369,7 @@ def cmake_args(self):
         # TODO: disable this once Cray-MPICH is fixed
         if (
             self.spec.satisfies("@4.2.00:")
-            and "mpi" in self.spec
+            and self.spec.satisfies("mpi")
             and self.spec["mpi"].name == "cray-mpich"
```

Comment:
Should be identical to:
```suggestion
            and self.spec.satisfies("^[virtuals=mpi] cray-mpich")
```